diff --git a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode.h b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode.h
index 37f0b7c8e..11e1625cc 100644
--- a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode.h
+++ b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode.h
@@ -1,3 +1,8 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
 /*
  * Generated using zcbor version 0.6.0
  * https://github.com/NordicSemiconductor/zcbor
diff --git a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode_types.h b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode_types.h
index c0e3599fb..46a7f6c02 100644
--- a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode_types.h
+++ b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_decode_types.h
@@ -1,3 +1,8 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
 /*
  * Generated using zcbor version 0.6.0
  * https://github.com/NordicSemiconductor/zcbor
diff --git a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode.h b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode.h
index 57fcfaf40..81e7639f3 100644
--- a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode.h
+++ b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode.h
@@ -1,3 +1,8 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
 /*
  * Generated using zcbor version 0.6.0
  * https://github.com/NordicSemiconductor/zcbor
diff --git a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode_types.h b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode_types.h
index f3dda00ce..5fe366c01 100644
--- a/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode_types.h
+++ b/subsys/net/lib/nrf_provisioning/include/nrf_provisioning_cbor_encode_types.h
@@ -1,3 +1,8 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
 /*
  * Generated using zcbor version 0.6.0
  * https://github.com/NordicSemiconductor/zcbor
diff --git a/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_decode.c b/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_decode.c
index 4a738032e..2af073385 100644
--- a/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_decode.c
+++ b/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_decode.c
@@ -1,3 +1,8 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
 /*
  * Generated using zcbor version 0.6.0
  * https://github.com/NordicSemiconductor/zcbor
diff --git a/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_encode.c b/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_encode.c
index ef0cb9b3c..35817df1f 100644
--- a/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_encode.c
+++ b/subsys/net/lib/nrf_provisioning/src/nrf_provisioning_cbor_encode.c
@@ -1,3 +1,8 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
 /*
  * Generated using zcbor version 0.6.0
  * https://github.com/NordicSemiconductor/zcbor
